pkg/agent/util/iptables/builder.go

```diff
@@ -67,20 +69,56 @@ func (b *iptablesRuleBuilder) MatchCIDRDst(cidr string) IPTablesRuleBuilder {
 	return b
 }
 
-func (b *iptablesRuleBuilder) MatchIPSetSrc(ipset string) IPTablesRuleBuilder {
-	if ipset == "" {
+func (b *iptablesRuleBuilder) MatchIPSrc(ip string) IPTablesRuleBuilder {
+	if ip == "" || ip == "0.0.0.0" || ip == "::" {
```

Comment:
I don't think this is really meaningful. For example, "0.0.0.0" doesn't usually mean "match any IP address". This is different from the meaning of CIDR "0.0.0.0/0".

IMO, we should not add these new methods (`MatchIPSrc`, `MatchIPDst`). I didn't see a compelling reason to do it. Instead of `MatchIPSrc("192.168.77.100")`, we can just do `MatchCIDRSrc("192.168.77.100/32")`, or even `MatchCIDRSrc("192.168.77.100")`. From an iptables perspective this is the same, and we don't do any input validation anyway?

Reply:
You are right. I have removed these two methods and added extra checks to skip `0.0.0.0` or `::` to `MatchCIDRSrc` and `MatchCIDRDst`.

Reply:
That's not exactly what I meant.  `"0.0.0.0"` and `"::"` don't usually mean "match any IP address of that family" AFIK. So I don't think we should handle them in a special way. If someone wants to explicitly match any IP for some reason, they should use the proper CIDR (`"0.0.0.0/0"` / `"::/0"`).

Reply:
My bad, `0.0.0.0` or `::` means "matching nonthin IP address`. Thanks for pointing out this. `0.0.0.0/0` or `::/0` should be used to match any IPv4 or IPv6 addresses.